storage/host-flash/host-flash.c

```diff
@@ -51,28 +52,37 @@ ssize_t hostflash_read(unsigned int addr, void *buff, size_t bufflen)
 
 ssize_t hostflash_write(unsigned int addr, void *buff, size_t bufflen)
 {
-    char currentByte = 0;
-    size_t wrote = 0;
+    char tempTab[256];
+    size_t wrote = 0, i;
+    int chunkLen;
     ssize_t stat;
 
     if (addr + bufflen > hostflash_common.flashsz)
         return -EINVAL;
 
     while (wrote != bufflen) {
-        stat = hostflash_read(addr + wrote, &currentByte, 1);
+        if ((bufflen - wrote) / sizeof(tempTab))
+            chunkLen = sizeof(tempTab);
+        else
+            chunkLen = (bufflen - wrote) % sizeof(tempTab);
```

Comment:
What do you think about defining size of `tempTab` as local const variable or define? Then you don't have to invoke to `sizeof` function each time.

Reply:
just FYI: sizeof is not a function, it's an operator which is resolved at compile time to constant value - so in terms of performance it won't get any faster.

I would advise to introduce intermediate value only if it improves readability, eg doing repeatedly:
```c
// multiple similar loops
for (unsigned int i = 0; i < sizeof(tab) / sizeof(tab[0]); ++i)
    // [...]
```
I would change it to
```c
const unsigned int tabLen =  sizeof(tab) / sizeof(tab[0]);

for (unsigned int i = 0; i < tabLen; ++i)
    // [...]
```

In the above case the performance would be exactly the same (const value constraint at compile time)

Reply:
OK